src/signs_gui.cpp

```diff
@@ -569,6 +569,10 @@ void HandleClickOnSign(const Sign *si)
 		RenameSign(si->index, "");
 		return;
 	}
+
+	/* If we can't rename the sign, don't even open the rename GUI. */
+	if (!CompanyCanRenameSign(si)) return;
```

Comment:
IMO it would be more logical to check this before the ctrl-click event though currently it doesn't matter as ctrl-click condition is more strict.